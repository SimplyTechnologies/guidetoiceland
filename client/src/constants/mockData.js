let _id = 0;

const genId = () => ++_id;

function generateMock() {
	return [{
		src: 'https://guidetoiceland.imgix.net/5753/x/0/62461501e91513e89581980bb7b06bf7-.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=b8d5aefbd39c6a3261a7906bf37086f6',
		id: genId(),
		description: `Iceland is spectacular in so many ways and Icelandic nature is quite unique with its vast landscape,  volcanic activity, geothermal areas, glacier lagoons and sceneries, black sand beaches and spectacular mountain ridges.  The country's most recognizable series of attractions though are maybe the endless amount of waterfalls making Iceland quite unique. Once I heard the number 10.000. `,
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/267234/x/0/agust-2014-austfirdir-019.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=aff79094054018fceff66d7684f6deed',
		id: genId(),
		description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer.',
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/5538/x/0/69263b425cbe576ed488ffa9ced7e522-.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=3d56f0abafbbb08150348108b7997195',
		id: genId(),
		description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer.',
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/9658/x/0/981407afbefdcd03617f300c9661daa4-.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=026862bb140c41e2dec19c460d6ac25a',
		id: genId(),
		description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer.',
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/196386/x/0/rsz-img-3054-2.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=745cbd78701713d013023cd1b08fcfb4',
		id: genId(),
		description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer.',
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/9782/x/0/598e91aafca4b7f7ff27179c37bfeb0a-.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=2f1041c4cd2e5a74078cd3114b09cce7',
		id: genId(),
		description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer.',
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/5753/x/0/62461501e91513e89581980bb7b06bf7-.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=b8d5aefbd39c6a3261a7906bf37086f6',
		id: genId(),
		description: `Iceland is spectacular in so many ways and Icelandic nature is quite unique with its vast landscape,  volcanic activity, geothermal areas, glacier lagoons and sceneries, black sand beaches and spectacular mountain ridges.  The country's most recognizable series of attractions though are maybe the endless amount of waterfalls making Iceland quite unique. Once I heard the number 10.000. `,
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/267234/x/0/agust-2014-austfirdir-019.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=aff79094054018fceff66d7684f6deed',
		id: genId(),
		description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer.',
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/5538/x/0/69263b425cbe576ed488ffa9ced7e522-.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=3d56f0abafbbb08150348108b7997195',
		id: genId(),
		description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer.',
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/9658/x/0/981407afbefdcd03617f300c9661daa4-.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=026862bb140c41e2dec19c460d6ac25a',
		id: genId(),
		description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer.',
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/196386/x/0/rsz-img-3054-2.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=745cbd78701713d013023cd1b08fcfb4',
		id: genId(),
		description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer.',
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/9782/x/0/598e91aafca4b7f7ff27179c37bfeb0a-.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=2f1041c4cd2e5a74078cd3114b09cce7',
		id: genId(),
		description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer.',
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/5753/x/0/62461501e91513e89581980bb7b06bf7-.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=b8d5aefbd39c6a3261a7906bf37086f6',
		id: genId(),
		description: `Iceland is spectacular in so many ways and Icelandic nature is quite unique with its vast landscape,  volcanic activity, geothermal areas, glacier lagoons and sceneries, black sand beaches and spectacular mountain ridges.  The country's most recognizable series of attractions though are maybe the endless amount of waterfalls making Iceland quite unique. Once I heard the number 10.000. `,
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/267234/x/0/agust-2014-austfirdir-019.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=aff79094054018fceff66d7684f6deed',
		id: genId(),
		description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer.',
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/5538/x/0/69263b425cbe576ed488ffa9ced7e522-.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=3d56f0abafbbb08150348108b7997195',
		id: genId(),
		description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer.',
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/9658/x/0/981407afbefdcd03617f300c9661daa4-.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=026862bb140c41e2dec19c460d6ac25a',
		id: genId(),
		description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer.',
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/196386/x/0/rsz-img-3054-2.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=745cbd78701713d013023cd1b08fcfb4',
		id: genId(),
		description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer.',
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/9782/x/0/598e91aafca4b7f7ff27179c37bfeb0a-.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=2f1041c4cd2e5a74078cd3114b09cce7',
		id: genId(),
		description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer.',
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/5753/x/0/62461501e91513e89581980bb7b06bf7-.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=b8d5aefbd39c6a3261a7906bf37086f6',
		id: genId(),
		description: `Iceland is spectacular in so many ways and Icelandic nature is quite unique with its vast landscape,  volcanic activity, geothermal areas, glacier lagoons and sceneries, black sand beaches and spectacular mountain ridges.  The country's most recognizable series of attractions though are maybe the endless amount of waterfalls making Iceland quite unique. Once I heard the number 10.000. `,
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/267234/x/0/agust-2014-austfirdir-019.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=aff79094054018fceff66d7684f6deed',
		id: genId(),
		description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer.',
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/5538/x/0/69263b425cbe576ed488ffa9ced7e522-.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=3d56f0abafbbb08150348108b7997195',
		id: genId(),
		description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer.',
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/9658/x/0/981407afbefdcd03617f300c9661daa4-.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=026862bb140c41e2dec19c460d6ac25a',
		id: genId(),
		description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer.',
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/196386/x/0/rsz-img-3054-2.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=745cbd78701713d013023cd1b08fcfb4',
		id: genId(),
		description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer.',
		title: 'Lorem ipsum ahmist ahad'
	}, {
		src: 'https://guidetoiceland.imgix.net/9782/x/0/598e91aafca4b7f7ff27179c37bfeb0a-.jpg?auto=compress%2Cformat&ch=Width%2CDPR&dpr=1&h=600&ixlib=php-1.1.0&w=300&s=2f1041c4cd2e5a74078cd3114b09cce7',
		id: genId(),
		description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry\'s standard dummy text ever since the 1500s, when an unknown printer.',
		title: 'Lorem ipsum ahmist ahad'
	}]
}

function shuffle(b) {
	let a = [...b];
    for (let i = a.length - 1; i > 0; i--) {
        const j = Math.floor(Math.random() * (i + 1));
        [a[i], a[j]] = [a[j], a[i]];
    }
    return a;
}


export const mockData = {
	'Popular tips': generateMock(),
	'Verified locals': shuffle(generateMock()),
	'Latest tips': shuffle(generateMock()),
	'Newest locals': shuffle(generateMock())
};