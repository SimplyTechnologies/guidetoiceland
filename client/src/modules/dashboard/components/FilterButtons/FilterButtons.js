import React from 'react';
import { Row, Col, Button, Container } from 'reactstrap';
import PropTypes from 'prop-types';
import { mockData } from 'constants/mockData';
import classNames from 'classnames';
const buttons = Object.keys(mockData);

const isButtonSelected = (selectedButton, btn) => selectedButton === btn;

const addButtonColor = (selectedButton, btn) =>
	isButtonSelected(selectedButton, btn)
		? 'success'
		: 'secondary'

function FilterButtons({
	onPress,
	selectedButton
}) {
	return (
		<Container>
			<Row>
				{buttons.map(btn => (
					<Col
						key={btn}
						size={4}
						className={'filter-buttons-col'}
					>
						<Button
							className={
								classNames({
									'button': true,
									'unselected-button': !isButtonSelected(selectedButton, btn),
									'selected-button': isButtonSelected(selectedButton, btn)
								})
							}
							onClick={() => onPress(btn)}
						>
							{btn}
						</Button>
					</Col>
				))}
			</Row>
		</Container>
	)
}

FilterButtons.propTypes = {
	onPress: PropTypes.func.isRequired,
	selectedButton: PropTypes.string.isRequired
}

export default FilterButtons;