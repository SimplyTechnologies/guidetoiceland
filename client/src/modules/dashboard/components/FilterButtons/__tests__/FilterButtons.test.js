import React from 'react';

import { mockData } from 'constants/mockData';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import FilterButtons from '../FilterButtons';

describe('FilterButtons', () => {
	let instance;
	let renderedInstance;
	let onPressFn = jest.fn();

	beforeEach(() => {
		onPressFn.mockReset();
		instance = mount(
			<FilterButtons
				selectedButton={Object.keys(mockData)[0]}
				onPress={onPressFn}
			/>
		);
		renderedInstance = renderer.create(
			<FilterButtons
				selectedButton={Object.keys(mockData)[0]}
				onPress={onPressFn}
			/>
		);
	})
	it('should render correctly', () => {
		expect(renderedInstance).toMatchSnapshot();
	});
	it('should highlight selected button', () => {
		expect(instance.find('button').first().hasClass('selected-button')).toBe(true)
	});
	it('should invoke onPress prop', () => {
		const button = instance.find('button').first();
		const selectedFilter = Object.keys(mockData)[0];
		button.simulate('click');

		expect(onPressFn).toHaveBeenCalledWith(selectedFilter);
	});
});