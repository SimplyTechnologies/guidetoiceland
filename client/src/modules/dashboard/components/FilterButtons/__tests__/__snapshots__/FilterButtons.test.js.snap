// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterButtons should render correctly 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="filter-buttons-col col"
      size={4}
    >
      <button
        className="button selected-button btn btn-secondary"
        onClick={[Function]}
        type="button"
      >
        Popular tips
      </button>
    </div>
    <div
      className="filter-buttons-col col"
      size={4}
    >
      <button
        className="button unselected-button btn btn-secondary"
        onClick={[Function]}
        type="button"
      >
        Verified locals
      </button>
    </div>
    <div
      className="filter-buttons-col col"
      size={4}
    >
      <button
        className="button unselected-button btn btn-secondary"
        onClick={[Function]}
        type="button"
      >
        Latest tips
      </button>
    </div>
    <div
      className="filter-buttons-col col"
      size={4}
    >
      <button
        className="button unselected-button btn btn-secondary"
        onClick={[Function]}
        type="button"
      >
        Newest locals
      </button>
    </div>
  </div>
</div>
`;
